lib/client/doc.js

```diff
@@ -522,12 +525,15 @@ Doc.prototype._otApply = function(op, source) {
     // that the snapshot only include updates from the particular op component
     // at the time of emission. Eliminating this would require rethinking how
     // such external bindings are implemented.
-    if (!source && this.type === types.defaultType && op.op.length > 1) {
+    if ( (this.type.shatter && op.op.length > 1) && 
```

Comment:
Considering removing `op.op.length > 1` check...

My reasons would be:

1. Can operations for custom OT types be other objects than arrays? If so this length check would cause an error.
2. Maybe an OT type operation has a length of one but due to its structure it can actually be shattered into smaller ops.

Doing so would call op.shatter on ops of length 1 or 0. Don't believe this should be an issue since the shatter method would be extremely fast in these cases and OT types should support these cases. Thoughts?